src/templates/plugin.routing.js

```diff
@@ -55,16 +55,29 @@ function localeRoute (route, locale) {
     const isDefaultLocale = locale === defaultLocale
     // if route has a path defined but no name, resolve full route using the path
     const isPrefixed =
-        // don't prefix default locale
-        !(isDefaultLocale && [STRATEGIES.PREFIX_EXCEPT_DEFAULT, STRATEGIES.PREFIX_AND_DEFAULT].includes(strategy)) &&
-        // no prefix for any language
-        !(strategy === STRATEGIES.NO_PREFIX) &&
-        // no prefix for different domains
-        !i18n.differentDomains
-
-    let path = (isPrefixed ? `/${locale}${route.path}` : route.path)
-    path = path.replace(/\/+$/, '') + (trailingSlash ? '/' : '') || '/'
-    localizedRoute.path = path
+      // don't prefix default locale
+      !(isDefaultLocale && [STRATEGIES.PREFIX_EXCEPT_DEFAULT, STRATEGIES.PREFIX_AND_DEFAULT].includes(strategy)) &&
+      // no prefix for any language
+      !(strategy === STRATEGIES.NO_PREFIX) &&
+      // no prefix for different domains
+      !i18n.differentDomains
+    const thisRoute = this.router.resolve(route.path).route
+    const routeName = this.getRouteBaseName(thisRoute)
+
+    if (isPrefixed && !routeName) {
+      let path = `/${locale}${route.path}`
+      path = path.replace(/\/+$/, '') + (trailingSlash ? '/' : '') || '/'
+      localizedRoute.path = path
+    } else {
+      localizedRoute.path = route.path.replace(/\/+$/, '') + (trailingSlash ? '/' : '') || '/'
+      if (routeName) {
+        localizedRoute.name = getLocaleRouteName(routeName, locale)
+        localizedRoute.params = thisRoute.params
+        localizedRoute.query = thisRoute.query
+        localizedRoute.hash = thisRoute.hash
+        return this.router.resolve(localizedRoute).route
```

Comment:
You can remove this line and let the line at the end of the function handle that.

Reply:
it's also done